dace/codegen/targets/cpp.py

```diff
@@ -1350,6 +1348,7 @@ def synchronize_streams(sdfg, dfg, state_id, node, scope_exit, callsite_stream,
         # If we are the ones to remove the last terminator, release memory
         terminators.remove(scope_exit)
         if len(terminators) == 0:
+            backend = common.get_gpu_backend()
```

Comment:
Why isn't this (and line 1374) just replacing line 1331? Just curious; do we expect to somehow handle streams on different cards (and architectures) in the future?

Reply:
Since getting the architecture is now a more involved process that can fail and raise exceptions, this change only triggers it if a GPU is actually used